### ISSUE TITLE
What are the Chapters tested for Finals?

### ISSUE THREAD

**Issue by nicolaineo** [nus-cs2030/2021-s2#226]
Hi everybody, I can't seem to find the list of topics tested for Finals.
Can anybody enlighten me please. Thank you!!!

**Comment by keithcharlesc** [comment]
Hi, if I'm not wrong it's till end of Lecture 11 with Lecture 12 being non-examinable, the topics list are on the module details page of LumiNUS

**Comment by xingweird** [comment]
Hey! Thought to add on as well that LazyList in Lecture 10 is non-examinable too :) All the best for finals everyone

**Comment by nicolaineo** [comment]
Thanks so much guys!

**Comment by nicolaineo** [comment]
> Hey! Thought to add on as well that LazyList in Lecture 10 is non-examinable too :) All the best for finals everyone

So Lazy can be tested but not LazyList?

**Comment by xingweird** [comment]
> > Hey! Thought to add on as well that LazyList in Lecture 10 is non-examinable too :) All the best for finals everyone
> 
> So Lazy can be tested but not LazyList?

From what I understand, yeah!

**Comment by keithcharlesc** [comment]
> > Hey! Thought to add on as well that LazyList in Lecture 10 is non-examinable too :) All the best for finals everyone
> 
> So Lazy can be tested but not LazyList?

Yup but if I'm not wrong LazyList is mainly just the implementation that covers the concept of Streams which the concepts should be tested

**Comment by Jjjetplane** [comment]
Up till lecture 11 with the exception of LazyList if I'm not wrong!

**Comment by Atem729** [comment]
> Up till lecture 11 with the exception of LazyList if I'm not wrong!

I think its this too !

**Comment by bandytan** [comment]
Hi, everything up including and up till lecture 11, excluding the second half of lecture 10 (Lazylist). And yes Lazy evaluation can be tested.

**Comment by Buwoo** [comment]
Just to add on, whilst finals coverage is up till lecture 11 excluding lazy list, I would recommend at least watching the first half of lecture 12, where Prof Henry covers completable futures in greater detail & with examples than whatever was solely covered in lecture 11

**Comment by mltnsia** [comment]
Hi, everything up to lecture 11, excluding the second half of lecture 10 (Lazylist)

**Comment by ydymegan** [comment]
Finals coverage is up till lecture 11 excluding lazy list.

**Comment by JWulaXia** [comment]
should be up to lec 11 and except lazylist from lec10, but can I check that do we need to understand the idea in lec12 or we can totally ignore that chapter?

**Comment by Martini-Chwang** [comment]
iirc up to lec 11 except lazylist. PYP had lazylist so can still give it a try if u wanna test ur understanding on it

**Comment by deniselam10** [comment]
You can refer to the LumiNUS announcement Prof Henry posted on 21/04!
